### PR TITLE
[Validator] Remove the pModule input for RunInternalValidator

### PR DESCRIPTION
**Pull request by python3kgae**
This change ensures that data validation occurs within the container itself, 
rather than relying on the module—especially since the module may be modified during container assembly.

Furthermore, simplifying the validator’s interface would be an added benefit.

### CODE REVIEW

File: tools/clang/tools/dxcvalidator/dxcvalidator.cpp
Comment (damyanp) [inline tools/clang/tools/dxcvalidator/dxcvalidator.cpp:64]: Does this mean that DxcValidatorFlags_ModuleOnly no longer does anything?

Comment (python3kgae) [reply tools/clang/tools/dxcvalidator/dxcvalidator.cpp:64]: No. DxcValidatorFlags_ModuleOnly will go runDxilModuleValidation now.

Comment (damyanp) [reply tools/clang/tools/dxcvalidator/dxcvalidator.cpp:64]: Ah, ok I thought that this change removed all instances of it.

**Review by damyanp — APPROVED** [review]
LGTM, but I suggest you wait for a review from someone more familiar with this area before merging this.

File: tools/clang/tools/dxcompiler/dxcompilerobj.cpp
Comment (pow2clk) [inline tools/clang/tools/dxcompiler/dxcompilerobj.cpp:89]: This is not (any longer?) called in this file. The predeclaration can be removed.

**Review by pow2clk — COMMENTED** [review]
Looks good overall. I think the prototype should be removed from dxccompilerobj.cpp and I think we shouldn't have two definitions of `DiagRestore` in DxilValidation.cpp. The redundant `IsDxilContainerLike` check you can leave in if removing it makes you nervous.

File: tools/clang/tools/dxcompiler/dxcompilerobj.cpp
Comment (python3kgae) [reply tools/clang/tools/dxcompiler/dxcompilerobj.cpp]: Removed.

File: tools/clang/tools/dxcvalidator/dxcvalidator.cpp
Comment (python3kgae) [reply tools/clang/tools/dxcvalidator/dxcvalidator.cpp]: Removed.

File: lib/HLSL/DxilValidation.cpp
Comment (python3kgae) [reply lib/HLSL/DxilValidation.cpp]: Didn't even notice there's another DiagRestore in the file. :)
Merged by change Ctx to pointer.

File: lib/HLSL/DxilValidation.cpp
Comment (python3kgae) [reply lib/HLSL/DxilValidation.cpp]: Merged by change Ctx to pointer.

**Review by pow2clk — APPROVED** [review]
Thanks!

**Pull request merged**
